### DISCUSSION TITLE
IVR Insights / Side Modal & Visualization

### DISCUSSION THREAD

**Discussion by kylegilbreath** [twilio-labs/paste#923] [Office hours] [answered]
**The specific questions I have are...**

- Needing a way to disclose a lot of information in a modal dialog while keeping context of the visualization. Was leaning toward using a side modal (if that is in the future plans 😉). If not, are there other ideas on how to do this?
- Visualizer will be custom built using the Highcharts charting library. Any guidance here to keep in line with the Paste look and feel?
- Will we have an expandable table row component by Q2 of next year?

**Background:** We have wrapped up a first round of research on some early wireframes for the IVR Insights project, and before moving to high fidelity, wanted to get your thoughts on a few areas that we may not have components yet built to support.
**Stage of design:** Prototype
**Research:** (in sign-up sheet)
**Links to mockups or prototype:** https://twilio.invisionapp.com/share/XDYZ05TNT2A#/436508157_IVR_Prototype-IVR_Insights-Visualizer-1

**Comment by serifluous** [comment, marked as answer]
> Needing a way to disclose a lot of information in a modal dialog while keeping context of the visualization. Was leaning toward using a side modal (if that is in the future plans 😉). If not, are there other ideas on how to do this?

I'd recommend using the [Non-Modal Dialog Primitive](https://paste.twilio.design/primitives/non-modal-dialog-primitive#non-modal-dialog-without-gutter) for now (which might actually be more usable than a Modal for your case if you want to keep the visualization in view) and style your own. I'd recommend copying the [shadow, border, and background of the Popover](https://paste.twilio.design/components/popover#popover). We don't have a Sketch component yet for Popover, but Ally's working on it.

I haven't actually chatted with @sang-yeo about this yet, but if you look at the Invision link he shared (same office hours sign-up day as yours), you can see something similar on the very last page.

![Screen Shot 2020-11-11 at 2 53 53 PM](https://user-images.githubusercontent.com/3821840/98873734-c0b2be80-242d-11eb-99ad-60019396f84c.png)


> Visualizer will be custom built using the Highcharts charting library. Any guidance here to keep in line with the Paste look and feel?

We have [data visualization colors coming in our next release](https://paste.twilio.design/roadmap)! You'll be able to plug the color values into Highcharts. The design work for this will likely kick off for this starting next sprint (Nov 18), so we can loop you/Travis in then. If you have any specific requirements beyond the colors needing to be distinguishable enough from each other, please let us know!

> Will we have an expandable table row component by Q2 of next year?

Nope, and we might not ever have anything exactly called "expandable table rows"...but! The way you've used the expandable sections in tables here is pretty much aligned with a Tree Grid component. Still won't be available by Q2 unfortunately because that needs to be built on top of a Data Grid, which is complex in itself. We've been recommending that folks use a Modal as a way to "expand" more data in a table row for now.
